cmd/fyne/internal/commands/build.go

```diff
@@ -6,16 +6,58 @@ import (
 	"runtime"
 	"strings"
 
+	version "github.com/mcuadros/go-version"
 	"golang.org/x/sys/execabs"
 )
 
+type runner interface {
+	RunOutput(arg ...string) ([]byte, error)
+}
+
+type command string
+
+func (c command) RunOutput(arg ...string) ([]byte, error) {
+	return execabs.Command(string(c), arg...).Output()
+}
+
+func runCommandOutput(runner runner, args ...string) ([]byte, error) {
+	return runner.RunOutput(args...)
+}
+
 type builder struct {
 	os, srcdir, target string
 	release            bool
 	tags               []string
 }
 
+func checkVersion(runner runner, versionConstraint *version.ConstraintGroup) error {
```

Comment:
The version could be also obtained using `runtime.Version()`. It could be good enough to check only against the minor version for now and avoid the introduction of a new dependency.
```suggestion
func checkVersion(minorConstraint int) (bool, error) {
	version := runtime.Version()
	minor, err := strconv.Atoi(version[4:6])
	if err != nil {
		return false, fmt.Errorf("invalid version: %s", version)
	}
	if minor < minorConstraint {
		return false, fmt.Errorf("expected Go minor version %d. Got %d from version %s", minor, minorConstraint, version)
	}
	return true, nil
}
```

Reply:
No, this won't work as there is no guaranty that the go installed is the same as the one used to compile the fyne command and so there is no link between runtime and compile time.

Reply:
Oh, right we are going to invoke the go binary directly... I've missed that sorry.
Btw, if there are no downside the check about the minor version to avoid the addition of the dependency could be still valid.

Reply:
This is a good point. Realistically all we are doing is checking for "17" at the beginning of the version string...

Reply:
The version parsing is sadly not so easy as people could be using a git release of go for example. I do not see any drawback into relying on someone else work here instead of reinventing the wheel. The dependency only impact fyne command. It is MIT, small, well tested, used and more future proof that what we could come up with.

Reply:
I see, but if I understood correctly, we need to check only if the minor version of the go binary used to compile in wasm is greater than `17` (this should be true until Go 2.0 is released). Looking at the code we are already extracting the version from the `go version` command output in a custom way so comparing directly the minor version should be good enough.

Anyway if we want to use the module, probably we would consider some alternatives like https://github.com/hashicorp/go-version since the one used seems archived.

Reply:
I did go with hashicorp go-version as it was MPL and not sure that would be ok for the project. If it is, I have no problem changing to it as they are pretty much equivalent.

Reply:
I don't think MPL is OK for us, copying my Slack message here:

Following discussion on tickets I reviewed MPL v2 and I don't think we can use it... The main problematic clause is 3.2.1: "If You distribute Covered Software in Executable Form then: [1] such Covered Software must also be made available in Source Code Form..." - we cannot put that requirement on our users.